sql/moz-fx-data-shared-prod/analysis/unified_metrics_v1/query.sql

```diff
@@ -0,0 +1,258 @@
+      -- expected to produce is 'Fenix BrowserStack'
+      IF(
+        isp = 'BrowserStack',
+        CONCAT(normalized_app_name, ' BrowserStack'),
+        normalized_app_name
+      ) AS normalized_app_name
+    )
+  FROM
+    unioned_source
+),
+search_clients AS (
+  SELECT
+    *
+  FROM
+    search_derived.mobile_search_clients_daily_v1
+  WHERE
+    submission_date = DATE_ADD(@submission_date, INTERVAL 1 DAY)
+),
+search_metrics AS (
+  SELECT
+    unioned.client_id,
+    unioned.submission_date,
+        -- the table is more than one row per client (one row per engine, looks like), so we have to aggregate.
+    SUM(ad_click) AS ad_click,
+    SUM(organic) AS organic_search_count,
+    SUM(search_count) AS search_count,
+    SUM(search_with_ads) AS search_with_ads,
+  FROM
+    unioned
+  LEFT JOIN
+    search_clients m
+  ON
+    unioned.client_id = m.client_id
+    AND DATE_ADD(unioned.submission_date, INTERVAL 1 DAY) = m.submission_date
+  GROUP BY
+    1,
+    2
+),
+mobile_with_searches AS (
+  SELECT
+    unioned.client_id,
+    unioned.sample_id,
+    CASE
+    WHEN
+      BIT_COUNT(days_seen_bits)
+      BETWEEN 1
+      AND 6
+    THEN
+      'infrequent_user'
+    WHEN
+      BIT_COUNT(days_seen_bits)
+      BETWEEN 7
+      AND 13
+    THEN
+      'casual_user'
+    WHEN
+      BIT_COUNT(days_seen_bits)
+      BETWEEN 14
+      AND 20
+    THEN
+      'regular_user'
+    WHEN
+      BIT_COUNT(days_seen_bits) >= 21
+    THEN
+      'core_user'
+    ELSE
+      'other'
+    END
+    AS activity_segment,
+    unioned.normalized_app_name,
+    unioned.app_display_version AS app_version,
+    unioned.normalized_channel,
+    unioned.country,
+    unioned.city,
+    unioned.days_seen_bits,
+    DATE_DIFF(unioned.submission_date, unioned.first_seen_date, DAY) AS days_since_first_seen,
+    unioned.device_model,
+    unioned.is_new_profile,
+    unioned.locale,
+    unioned.first_seen_date,
+    unioned.days_since_seen,
+    unioned.normalized_os,
+    unioned.normalized_os_version,
+    COALESCE(
+      CAST(NULLIF(SPLIT(unioned.normalized_os_version, ".")[SAFE_OFFSET(0)], "") AS INTEGER),
+      0
+    ) AS os_version_major,
+    COALESCE(
+      CAST(NULLIF(SPLIT(unioned.normalized_os_version, ".")[SAFE_OFFSET(1)], "") AS INTEGER),
+      0
+    ) AS os_version_minor,
+    COALESCE(
+      CAST(NULLIF(SPLIT(unioned.normalized_os_version, ".")[SAFE_OFFSET(2)], "") AS INTEGER),
+      0
+    ) AS os_version_patch,
+    unioned.durations,
+    unioned.submission_date,
+    unioned.uri_count,
+    unioned.is_default_browser,
+    unioned.distribution_id AS distribution_id,
+    CAST(NULL AS string) AS attribution_content,
+    CAST(NULL AS string) AS attribution_source,
+    CAST(NULL AS string) AS attribution_medium,
+    CAST(NULL AS string) AS attribution_campaign,
+    CAST(NULL AS string) AS attribution_experiment,
+    CAST(NULL AS string) AS attribution_variation,
```

Comment:
```suggestion
    NULL AS attribution_variation,
```

Reply:
The CAST is required for compatibility, unfortunately only NULL can't lead to type mismatch when doing the UNION.